### DISCUSSION TITLE
Unable to retrieve channel by id: edge not found

### DISCUSSION THREAD

**Discussion by ConorOkus** [lightningdevkit/rust-lightning#2031] [Help] [answered]
I'm having issues receiving inbound payments, currently have an LDK node with a private channel open to an LND node. 

**Environment**
Regtest/Polar 
LDK Kotlin bindings version v0.0.113.0
LND version 0.15.5-beta

Example invoice [here](https://lightningdecoder.com/lnbcrt100u1p374yysdq2vdhkven9v5np4qwx7ugwljvumxu6tda0vrnwqus2mds0h7et9pvt6t35qcvptat6pxpp5q4mxnkmr5gpe2yhgkwgde23hmsqfr6tna7paat53ap4jyjwqa6vqsp5lk297y4ldwj7d8syd5zwdyrtkc4snnd8atsdaqqshma9hx9c6xpq9qyysgqcqpcxqzfvgtap26t4fkm08pnfhlt45fjvnjuvtrzx24cx3kp9lkjmev8j4n84yq0p5g2mqvjtytj83uvqq4z2ujl7s80dkq9ac7fl30u5ccysvfsqrsxez4)

LND Node logs
```
2023-02-13 19:33:51.257 [ERR] CRTR: Unable to retrieve channel by id: edge not found
2023-02-13 19:33:51.267 [INF] CRTR: Payment 057669db63a2039512e8b390dcaa37dc0091e973ef83deae91e86b2249c0ee98 failed: final_outcome=error, raw_err=TemporaryChannelFailure(update=(*lnwire.ChannelUpdate)(0xc001cef970)({
 Signature: (lnwire.Sig) (len=64 cap=64) {
  00000000  64 68 94 60 54 dd 47 31  b9 88 bc ef b1 29 cc f8  |dh.`T.G1.....)..|
  00000010  5a ea a5 44 12 51 4c 1d  a9 4c db 6b 25 c9 4e 0b  |Z..D.QL..L.k%.N.|
  00000020  41 c0 02 44 ea 42 7a 57  43 08 ae ce 29 e2 3c 19  |A..D.BzWC...).<.|
  00000030  85 6d 66 89 c6 f7 2a 08  b2 18 bc b0 cb 78 b6 41  |.mf...*......x.A|
 },
 ChainHash: (chainhash.Hash) (len=32 cap=32) 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206,
 ShortChannelID: (lnwire.ShortChannelID) 110:110:0,
 Timestamp: (uint32) 1676315169,
 MessageFlags: (lnwire.ChanUpdateMsgFlags) 00000001,
 ChannelFlags: (lnwire.ChanUpdateChanFlags) 00000001,
 TimeLockDelta: (uint16) 72,
 HtlcMinimumMsat: (lnwire.MilliSatoshi) 1 mSAT,
 BaseFee: (uint32) 1000,
 FeeRate: (uint32) 0,
 HtlcMaximumMsat: (lnwire.MilliSatoshi) 90000000 mSAT,
 ExtraOpaqueData: (lnwire.ExtraOpaqueData) {
 }
})
```

**Comment by TheBlueMatt** [comment, marked as answer]
I don't know lnd error codes super well, but it looks like your invoice didn't contain any route hints and lnd is saying it doesn't know where to forward the channel to. However, whoever sent the payment *did* know how to forward it (I guess maybe you had a public channel that isn't closed yet or something?) cause it had an SCID to forward on to in the onion? Can you give a bit more detail about your setup and maybe the LDK logs?

**Reply by ConorOkus** [reply]
I tend to get a bit confused with the logs output, probably because I have it logging everything [here](https://github.com/ConorOkus/uMlando-wallet/blob/master/app/src/main/java/com/example/umlandowallet/Start.kt#L180). Is there anything I should filter for specifically? 

I only have a single private channel open between myself and the LND node.  

I'm able to make successful payments
```
2023-02-13 18:27:37.246 3418-3663/com.example.umlandowallet I/LDKTAG: Scoring path through SCID 119846774571009 as having succeeded at 100000000 msat.
2023-02-13 18:27:37.247 3418-3663/com.example.umlandowallet I/LDKTAG: Not able to learn for channel with SCID 119846774571009 as we do not have graph info for it (likely a route-hint last-hop).
```

Some logs [here](https://pastebin.com/q4dHHTMa) it's a bit of a dump, so sorry if not helpful.

**Reply by TheBlueMatt** [reply]
> I tend to get a bit confused with the logs output, probably because I have it logging everything [here](https://github.com/ConorOkus/uMlando-wallet/blob/master/app/src/main/java/com/example/umlandowallet/Start.kt#L180). Is there anything I should filter for specifically?

I'm happy to look at all the logs :)

Your issue is at https://github.com/ConorOkus/uMlando-wallet/blob/794ce3396fd6eef55d5da659efce5c0a7eed62f0/app/src/main/java/com/example/umlandowallet/data/remote/AccessImpl.kt#L89-L99 - the first parameter to those `of` calls should be the transaction's position within the block, not the block height. Judging from the lnd side's SCID it should be 1 (i.e.e the first transaction after the coinbase transaction), but your side is sending an SCID of 109:109:0 (it looks like in your original post it was 110:110:0).

**Reply by ConorOkus** [reply]
Thanks.

I've updated that param to now take the `merkle_proof_pos` [here](https://github.com/ConorOkus/uMlando-wallet/pull/34/files#diff-1adcddad549fde3fd5220682e24cf9e4b7b02dee80e1c78a52a39dcada1a9dddR88)

Seems to be getting the correct SCID now.
```
2023-02-16 20:25:13.370 [WRN] CRTR: Attempt 1 for payment da5148b6d1163d1ada49d348d5f99c61a68c4c2b064a487f62c34c2e605ddba7 failed: TemporaryChannelFailure(update=(*lnwire.ChannelUpdate)(0xc0000cd6b0)({
 Signature: (lnwire.Sig) (len=64 cap=64) {
  00000000  2d 0d be a8 ba 0a bf 68  c0 3b 33 ca d3 33 93 d8  |-......h.;3..3..|
  00000010  48 a5 c9 cc c8 ce 80 03  03 b4 8f da 59 44 5a bf  |H...........YDZ.|
  00000020  4f 40 38 ab 60 e7 9d 43  a5 3e b6 97 17 26 c4 22  |O@8.`..C.>...&."|
  00000030  1d 36 9d f8 c9 5f 1f a0  47 e6 cb 43 3e 8e f6 b1  |.6..._..G..C>...|
 },
 ChainHash: (chainhash.Hash) (len=32 cap=32) 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206,
 ShortChannelID: (lnwire.ShortChannelID) 109:2:1,
 Timestamp: (uint32) 1676578113,
 MessageFlags: (lnwire.ChanUpdateMsgFlags) 00000001,
 ChannelFlags: (lnwire.ChanUpdateChanFlags) 00000001,
 TimeLockDelta: (uint16) 72,
 HtlcMinimumMsat: (lnwire.MilliSatoshi) 1 mSAT,
 BaseFee: (uint32) 1000,
 FeeRate: (uint32) 0,
 HtlcMaximumMsat: (lnwire.MilliSatoshi) 900000000 mSAT,
 ExtraOpaqueData: (lnwire.ExtraOpaqueData) {
 }
})
)@1
2023-02-16 20:25:13.386 [INF] CRTR: Payment da5148b6d1163d1ada49d348d5f99c61a68c4c2b064a487f62c34c2e605ddba7 failed: final_outcome=error, raw_err=TemporaryChannelFailure(update=(*lnwire.ChannelUpdate)(0xc0000cd6b0)({
 Signature: (lnwire.Sig) (len=64 cap=64) {
  00000000  2d 0d be a8 ba 0a bf 68  c0 3b 33 ca d3 33 93 d8  |-......h.;3..3..|
  00000010  48 a5 c9 cc c8 ce 80 03  03 b4 8f da 59 44 5a bf  |H...........YDZ.|
  00000020  4f 40 38 ab 60 e7 9d 43  a5 3e b6 97 17 26 c4 22  |O@8.`..C.>...&."|
  00000030  1d 36 9d f8 c9 5f 1f a0  47 e6 cb 43 3e 8e f6 b1  |.6..._..G..C>...|
 },
 ChainHash: (chainhash.Hash) (len=32 cap=32) 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206,
 ShortChannelID: (lnwire.ShortChannelID) 109:2:1,
 Timestamp: (uint32) 1676578113,
 MessageFlags: (lnwire.ChanUpdateMsgFlags) 00000001,
 ChannelFlags: (lnwire.ChanUpdateChanFlags) 00000001,
 TimeLockDelta: (uint16) 72,
 HtlcMinimumMsat: (lnwire.MilliSatoshi) 1 mSAT,
 BaseFee: (uint32) 1000,
 FeeRate: (uint32) 0,
 HtlcMaximumMsat: (lnwire.MilliSatoshi) 900000000 mSAT,
 ExtraOpaqueData: (lnwire.ExtraOpaqueData) {
 }
})
)@1
2023-02-16 20:25:13.402 [WRN] RPCS: Unable to send payment: error
```

More LDK logs [here](https://pastebin.com/jH64AXY1)

**Reply by TheBlueMatt** [reply]
`TemporaryChannelFailure` probably implies there's not enough liquidity? Can you share the log output from whichever node is failing it?

**Reply by ConorOkus** [reply]
This was a liquidity issue in the end. I got confused switching between btc, sats and mSats and was sending the wrong amounts essentially violating the channel reserve rule. 

Posting a link to our stack exchange topic on this subject for anyone who may read this in the future.

https://bitcoin.stackexchange.com/questions/116743/i-opened-a-channel-and-sent-1500-msats-to-my-counterparty-why-can-i-still-not-r

Thanks Matt
